### PR TITLE
Add navigation structure for learn section

### PR DESCRIPTION
**Pull request by LKedward**
As we expand the Quickstart tutorial and others start contributing tutorial content, we need a central structure for organising the learning pages on the site.

I've drafted a possible solution in this PR:
- A new page layout is introduced with a navigation side bar which displays the contents of the learn section and the structure of the current page headings.
- A yaml file is used to define the contents and order of the navigation bar whereby learning pages are grouped into named chapters

This approach:
- makes navigation explicitly clear to site users (no getting lost among lots of pages);
- allows contributors to start adding tutorial pages to the site now;
- but is flexible enough that it can easily be restructured as the number of tutorials increases;

To add a page, contributors need only write a markdown file and add an entry under the appropriate chapter in the yaml data file. (I will update CONTRIBUTING.md)

For this draft PR I've just included the existing quickstart tutorial to demonstrate; eventually the landing page will be updated to include useful links etc.

@milancurcic , @certik what are your thoughts?

#build_preview

### CODE REVIEW

**Comment by LKedward** [comment]
#build_preview

**Comment by milancurcic** [comment]
I think this is great and a step forward. I approve. 

I anticipate that our structure of content will be more unordered than ordered, so we may drop the numbering, but we can figure this out later as we go.

**Comment by LKedward** [comment]
Okay, I've restructured this PR so that it's less like a single monolithic book with chapters and
instead it allows single-page and multi-page mini-book tutorials.

The name and contents of each mini-book is defined in a yaml data file which is used to: populate the Learn landing page with a list of books, grouped by category; and populate a table-of-contents for each mini-book.

To demonstrate this, the Quickstart tutorial has been broken out into multiple pages so it's easier to navigate.

The 'Learn' landing page has been updated to also include an initial selection of third-party resources as well.

The Learn page is a little bare at the moment, but with the structure now in place it should be quite easy to start adding more mini-book tutorials (starting with #90).
I will add a section to the contributor guide on how to write a mini-book tutorial.

#build_preview

**Comment by certik** [comment]
@LKedward this looks really great. Lots of work done here thank you!

We should put there a link to https://github.com/j3-fortran/fortran_proposals also I think.

I think it looks great, +1 from me.

File: _data/learning.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    location: Oxford, UK
+    publisher: Oxford Universtiy Press
+
+  - author: Curcic, M.
+    year: 2019
+    title: "Modern Fortran: Building efficient parallel programs"
+    url: https://www.manning.com/books/modern-fortran
+    edition: 1st ed
+    publisher: Manning publications
+
```
Comment (milancurcic) [inline _data/learning.yml:79]:
```suggestion

  - author: Markus, A.
    year: 2012
    title: "Modern Fortran in Practice"
    url: https://www.cambridge.org/core/books/modern-fortran-in-practice/BC5BD23B2E478B4D457C5D6265BA9363
    edition: 1st ed
    location: Cambridge, UK
    publisher: Cambridge University Press
    
```

File: _data/learning.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    url: https://github.com/scivision/fortran2018-examples
+    description: A github repository containing code samples for various Fortran 2018 features
+
+
+
+
+# Print books listed at the bottom of the 'Learn' landing page
+#
+reference-books:
+  - author: Michael Metcalf, John Reid, and Malcolm Cohen
```
Comment (milancurcic) [inline _data/learning.yml:65]:
```suggestion
  - author: Metcalf, M., Reid, J. and Cohen M.
```

Comment (milancurcic) [reply _data/learning.yml:65]: I made this suggestion so the names are formatted the same way as other entries. But you could also list full names for all books, which is actually my preference.

**Comment by LKedward** [comment]
> An issue that comes up here is how to sort the books? Currently they don't seem sorted in any particular way. Books at the top of the list will be implicitly given higher priority or sense of importance. So we need to decide how to sort them:
> 
> * Alphabetically by title?
> * Alphabetically by author last name?
> * Chronologically from newest to oldest?

As yes, good point, I hadn't considered this.
I'm not sure what the best/standard way to sort them is - my preference would be to sort by date with newest first to prioritise up-to-date editions.

**Comment by LKedward** [comment]
Print books are now sorted by year descending - this can be changed to alphabetical if desired.
#build_preview

**Comment by jvdp1** [comment]
This looks really great. Easy to navigate and read! Really good job!

**Comment by LKedward** [comment]
Thanks all for the feedback and suggestions. I will now merge and work on updating the contributor guide with instructions in a new PR.

**Comment by LKedward** [comment]
#delete_preview

**Pull request merged**